Wash 929 write data record

Mrhea opened the pull request:
Tested and worked with local Tozny platform exposed over ngrok, using TozStore client credentials in the `onWriteRecordButtonPress` method.

Commit:
[WASH-929] Write record for iOS plugin

- Adds write record method for iOS plugin
- Updates E3dbSerializer methods to handle serializing
  records correctly
- Adds a new Dart method that runs only write record method for
  testing

Commit:
[WASH-929] Add serialization for file meta

- Adds fileMetaToJson method in Flutter
- Sets the onPressed value in Dart widget to onButtonPress

galxy25 commented on ios/Classes/E3dbSerializer.swift:
```diff
@@ -22,34 +22,51 @@ public class E3dbSerializer {
+        }
+        var data: [String: Any] = [:]
+        
+        /// If meta is not `nil`, then assume we can force unwrap its values
```
excellent comment

galxy25 commented on ios/Classes/SwiftFlutterPlugin.swift:
```diff
@@ -14,8 +14,56 @@ public class SwiftFlutterPlugin: NSObject, Flutter.FlutterPlugin {
+        let recordType = args["type"] as! String
+        let recordData: RecordData = RecordData(cleartext: fields)
+
+        let group = DispatchGroup()
```
some comment here that explains the hard earned knowledge around concurrency patterns for flutter to swift code for future readers would be much appreciated

galxy25 reviewed (approved):
:disco-gopher:

Commit:
[WASH-929] Document concurrency patterns

- Adds documentation for why a dispatch queue was used to
  solve a concurrency issue that caused a serialization error
